.pipelines/templates/.builder-release-template-windows.yaml

```diff
@@ -73,35 +86,73 @@ steps:
       -e ARCHITECTURE=${ARCHITECTURE} \
       ${CONTAINER_IMAGE} make -f packer.mk test-building-vhd
     displayName: Run VHD cache test
+
   - task: PublishPipelineArtifact@0
     inputs:
         artifactName: 'vhd-release-notes-${{ parameters.artifactName }}'
         targetPath: 'release-notes.txt'
+
   - script: |
       echo MODE=$(MODE) && \
       OS_DISK_SAS="$(cat packer-output | grep -a "OSDiskUriReadOnlySas:" | cut -d " " -f 2)" && \
       VHD_NAME="$(echo $OS_DISK_SAS | cut -d "/" -f 8 | cut -d "?" -f 1)" && \
       docker run --rm \
       -v ${PWD}:/go/src/github.com/Azure/AgentBaker \
       -w /go/src/github.com/Azure/AgentBaker \
-      -e CLIENT_ID=${CLIENT_ID} \
-      -e CLIENT_SECRET="$(CLIENT_SECRET)" \
-      -e TENANT_ID=${TENANT_ID} \
-      -e SUBSCRIPTION_ID="${SUBSCRIPTION_ID}" \
+      -e CLIENT_ID=${BUILD_CLIENT_ID} \
+      -e CLIENT_SECRET="$(BUILD_CLIENT_SECRET)" \
+      -e TENANT_ID=${BUILD_TENANT_ID} \
+      -e SUBSCRIPTION_ID="${BUILD_SUBSCRIPTION_ID}" \
+      -e AZURE_RESOURCE_GROUP_NAME=${BUILD_AZURE_RESOURCE_GROUP_NAME} \
+      -e AZURE_LOCATION=${BUILD_AZURE_LOCATION} \
       -e CLASSIC_BLOB=${STORAGE_ACCT_BLOB_URL} \
       -e CLASSIC_SAS_TOKEN="$(STORAGE_ACCT_SAS_TOKEN)" \
       -e OS_DISK_SAS=${OS_DISK_SAS} \
       -e VHD_NAME=${VHD_NAME} \
       ${CONTAINER_IMAGE} make -f packer.mk az-copy
     displayName: Copying resource to Classic Storage Account
     condition: and(succeeded(), eq(variables.DRY_RUN, 'False'), eq(variables['MODE'], 'default'))
+
+# Moved this conversion before cleanup because packer-generated SIG is already in VHD, and so can be cleaned.  
```

Comment:
`conversion` => `step`? How to understand `packer-generated SIG is already in VHD`?

Reply:
Moved conversion to VHD before cleanup as suggested. Because packer-generated SIG has already been converted to classical storage account, it can be deleted in cleanup for gen2Mode. For gen2SigMode (the one you suggested), this conversion will not be stepped in.

Reply:
Does the built SIG need to be deleted for gen2Mode (either Linux or Windows). I think if DRY_RUN=False, and the conversion to VHD is done, we don't need to keep the SIG for test? (This conversion step is after testing the built VHD)

Reply:
Synced in teams. The built SIG will be deleted if it's for production ``GEN2_SIG_FOR_PRODUCTION = True`` because ``SIG=>VHD``. Will not delete if the sig is only for test.